app/src/main/java/org/dhis2/utils/customviews/OptionSetPopUp.java

```diff
@@ -96,6 +95,7 @@ public OptionSetPopUp(Context context, View anchor, SpinnerViewModel model,
 
     public void setOptions(List<Option> options) {
         optionsMap = new HashMap<>();
+        Collections.sort(options, (option1, option2) -> option1.sortOrder().compareTo(option2.sortOrder()));
```

Comment:
do you think we could move the sort to another class and unit test it?

Reply:
We could, we should discuss it with the team the best way to this re-factorization. We also need to organize the customviews package better. 😕